app/src/main/java/com/dcrandroid/activities/more/PoliteiaActivity.kt

```diff
@@ -0,0 +1,854 @@
+package com.dcrandroid.activities.more
+
+import android.os.Bundle
+import android.os.Handler
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.ArrayAdapter
+import android.widget.TextView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.dcrandroid.R
+import com.dcrandroid.activities.BaseActivity
+import com.dcrandroid.adapter.ProposalAdapter
+import com.dcrandroid.data.Constants
+import com.dcrandroid.data.Proposal
+import com.dcrandroid.extensions.hide
+import com.dcrandroid.extensions.show
+import com.dcrandroid.util.Deserializer
+import com.google.gson.GsonBuilder
+import kotlinx.android.synthetic.main.activity_politeia.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.json.JSONObject
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.collections.ArrayList
+
+class PoliteiaActivity : BaseActivity(), SwipeRefreshLayout.OnRefreshListener, OnItemSelectedListener, ViewTreeObserver.OnScrollChangedListener {
+    private var proposals = ArrayList<Proposal>()
+    private var recyclerView: RecyclerView? = null
+    private var proposalAdapter: ProposalAdapter? = null
+    private var swipeRefreshLayout: SwipeRefreshLayout? = null
+    private var layoutManager: LinearLayoutManager? = null
+    private lateinit var emptyList: TextView
```

Comment:
Remove variable declarations for views and access them directing using their ids.